### PR TITLE
Windows hostnetwork alpha

### PR DESCRIPTION
**Pull request by marosset**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Allows kubelet to specific if Windows pods should be joined to the node or pod network namespace for Windows pods

#### Which issue(s) this PR fixes:

Part of: https://github.com/kubernetes/enhancements/issues/3503

#### Does this PR introduce a user-facing change?

```release-note
Adding alpha support for WindowsHostNetworking feature
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
[KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/3503-host-network-support-for-windows-pods
```

/sig windows node
/area kubelet

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_sandbox.go [hunk 1]
```diff
@@ -233,6 +233,15 @@ func (m *kubeGenericRuntimeManager) generatePodSandboxWindowsConfig(pod *v1.Pod)
 		SecurityContext: &runtimeapi.WindowsSandboxSecurityContext{},
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.WindowsHostNetwork) {
```
Comment (aojea) [inline pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:236]: just curiosity, what is the relation with `features.WindowsHostProcessContainers`?

are all `WindowsHostProcessContainers` HostNetwork pods?

Comment (marosset) [reply pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:236]: Yes, All pods that contain HostProcessContainers must set `hostNetwork=true` (this is enforced at API admission).
Really HostProcessContainers just start processes on the host but they are packages and deployed like contianers (hence the name).

With these changes pods that do not contain HostProcessContainers will have the option of using pod or node network namespace.

**Comment by marosset** [comment]
I have changes mostly working based on https://github.com/marosset/containerd/tree/windows-host-network-support so I think we can get this merged into K8s for v1.26!

/assign @jsturtevant @mrunalp @jayunit100

File: staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto [hunk 2]
```diff
@@ -935,6 +945,9 @@ message WindowsContainerSecurityContext {
 
     // Indicates whether a container is to be run as a HostProcess container.
     bool host_process = 3;
+
```
Comment (marosset) [inline staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto:948]: We don't need these changes - I will drop them.

**Comment by derekwaynecarr** [comment]
This PR looks fine.

Noting per the enhancement, we will need e2e tests, but you need the CRI changes merged from this PR to make that possible.

/approve
/lgtm

**Pull request merged**